tests/sentry/spans/test_buffer_v2.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import pytest
+from sentry_redis_tools.clients import StrictRedis
+
+from sentry.spans.buffer_v2 import RedisSpansBufferV2, Span
+
+
+@pytest.fixture
+def buffer():
+    return RedisSpansBufferV2()
+
+
+def assert_ttls(client: StrictRedis[bytes]):
+    """
+    Check that all keys have a TTL, because if the consumer dies before
+    flushing, we should not leak memory.
+    """
+
+    for k in client.keys("*"):
+        assert client.ttl(k) > -1, k
+
+
+def assert_clean(client: StrictRedis[bytes]):
+    """
+    Check that there's no leakage.
+
+    Note: CANNOT be done in pytest fixture as that one runs _after_ redis gets
+    wiped by the test harness.
+    """
+    assert not client.keys("*")
+
+
+def test_basic(buffer: RedisSpansBufferV2):
+    spans = [
+        Span(
+            payload=b"A", trace_id="a" * 32, span_id="c" * 16, parent_span_id="b" * 16, project_id=1
+        ),
+        Span(
+            payload=b"B", trace_id="a" * 32, span_id="d" * 16, parent_span_id="b" * 16, project_id=1
+        ),
+        Span(
+            payload=b"C", trace_id="a" * 32, span_id="e" * 16, parent_span_id="b" * 16, project_id=1
+        ),
+        Span(payload=b"D", trace_id="a" * 32, span_id="b" * 16, parent_span_id=None, project_id=1),
+    ]
+
+    buffer.process_spans(spans, now=0)
+
+    assert_ttls(buffer.client)
+
+    assert buffer.flush_segments(now=5) == {}
+    rv = buffer.flush_segments(now=11)
+    assert rv == {buffer._segment_id(1, "a" * 32, "b" * 16): {b"D", b"B", b"A", b"C"}}
+    assert buffer.flush_segments(now=30) == {}
```

Comment:
I am guessing that what's going on here is:
1. `flush_segments(now=5)` is a no-op because `span_buffer_root_timeout` is set to its default value of `10` and 5 is less than 10.
2. `flush_segments(now=11)` flushes segments because 11 is greater than 10.
3. `flush_segments(now=30)` is a no-op because all the segments are already flushed.

Am I correct? Either way, it would be nice for the test to be more self-evident for the reader. A brief comment would likely do the trick.